posthog/api/test/test_ingestion_warnings.py

```diff
@@ -0,0 +1,107 @@
+from posthog.kafka_client.topics import KAFKA_INGESTION_WARNINGS
+from posthog.models.event.util import format_clickhouse_timestamp
+from posthog.models.ingestion_warnings.sql import INSERT_INGESTION_WARNING
+from posthog.models.organization import Organization
+from posthog.test.base import APIBaseTest, ClickhouseTestMixin
+from posthog.utils import cast_timestamp_or_now
+
+
+def create_ingestion_warning(team_id: int, type: str, details: Dict, timestamp: str, source=""):
+    timestamp = cast_timestamp_or_now(timestamp)
+    data = {
+        "team_id": team_id,
+        "type": type,
+        "source": source,
+        "details": json.dumps(details),
+        "timestamp": format_clickhouse_timestamp(timestamp),
+        "_timestamp": format_clickhouse_timestamp(timestamp),
+    }
+    p = ClickhouseProducer()
+    p.produce(topic=KAFKA_INGESTION_WARNINGS, sql=INSERT_INGESTION_WARNING, data=data)
+
+
+class TestIngestionWarningsAPI(ClickhouseTestMixin, APIBaseTest):
+    @freeze_time("2021-12-04T19:20:00Z")
+    def test_ingestion_warnings_api(self):
+        create_ingestion_warning(
+            team_id=self.team.id,
+            type="cannot_merge_already_identified",
+            details={
+                "sourcePerson": "x-uuid",
+                "sourcePersonDistinctId": "Alice",
+                "targetPerson": "y-uuid",
+                "targetPersonDistinctId": "Bob",
+            },
+            timestamp="2021-12-03T00:00:00Z",
+        )
+        create_ingestion_warning(
+            team_id=self.team.id,
+            type="cannot_merge_already_identified",
+            details={},
+            timestamp="2021-12-02T00:00:00Z",
+        )
+        create_ingestion_warning(
+            team_id=self.team.id,
+            type="another_type",
+            details={},
+            timestamp="2021-11-15T00:00:00Z",
+        )
+        create_ingestion_warning(
+            team_id=self.team.id,
+            type="too_old",
+            details={},
+            timestamp="2021-11-01T00:00:00Z",
+        )
+
+        team2 = Organization.objects.bootstrap(None)[2]
+        create_ingestion_warning(
+            team_id=team2.id,
+            type="too_old",
+            details={},
+            timestamp="2021-12-01T00:00:00Z",
+        )
+
+        response = self.client.get(f"/api/projects/{self.team.pk}/ingestion_warnings")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            response.json(),
+            {
+                "results": [
+                    {
+                        "type": "cannot_merge_already_identified",
+                        "lastSeen": "2021-12-03T00:00:00Z",
+                        "warnings": [
+                            {
+                                "type": "cannot_merge_already_identified",
+                                "timestamp": "2021-12-03T00:00:00Z",
+                                "details": {
```

Comment:
If you check Karl's frontend pr, the the `type` would be written out & details displayed as JSON, which essentially accomplishes this goal.

Reply:
> can just call captureIngestionWarning with a reasonable message and be done with it?

We should definetly _not_ be doing that.

Technically the problem is:
1. we'd be stuck with whatever description we end up with in that code forever
2. with e.g. no way to link to persons or other tabs

Instead, we should be controlled with what we capture with `captureIngestionWarning` in the first place and make sure that:
1. It's actually useful for the customers and actionable
2. It's clearly documented what's going wrong

Not having all places instrumented with `captureIngestionWarning` if preferable to adding non-actionable noise for our users.

Reply:
Since we can filter out by type in the api when we send them, we don't need to be that careful as long as we always use unique types for warning locations added.

Reply:
If the only purpose of this is showing it to the user and giving them actionable advice, what does capturing extra events give us?

Also note a problem with `details`: the `details` object needs to be closely coupled with what we show in the FE (e.g. to link correctly) and hence needs to be properly thought through when we add an event.